subsys/shell/backends/Kconfig.backends

```
@@ -347,6 +347,13 @@ config SHELL_TELNET_SUPPORT_COMMAND
 	  Current support is so limited it's not interesting to enable it.
 	  However, if proven to be needed at some point, it will be possible
 	  to extend such support.
+	  It does have support for ECHO commands. When a telnet client sends
+	  a "do ECHO" command ("character at a time" mode), the telnet server
+	  will reply with "will ECHO" and start echoing each character sent
+	  to the server. This will give access to arrows and other escaped
+	  character commands.
```

Suggestion: My guess is that the maintainers would want a less "in-the-weeds" description, perhaps more like:
```suggestion
      to extend such support. It does have support for echo and "character
      at a time" mode, which enable the history and line-editing features
      of the shell.
```

---

sure